pkg/cloudclient/aws/private.go

```diff
@@ -648,6 +648,27 @@ func (c *Client) upsertARecord(clusterDomain, DNSName, aliasDNSZoneID, resourceR
 	if err != nil {
 		return err
 	}
+
+	var recordExists bool
+	input := &route53.ListResourceRecordSetsInput{
+		HostedZoneId: aws.String(publicHostedZoneID),
+	}
+
+	// Check if the Record we are tring to UPSERT already exists
+	err = c.route53Client.ListResourceRecordSetsPages(input, func(p *route53.ListResourceRecordSetsOutput, lastPage bool) bool {
+		for _, record := range p.ResourceRecordSets {
+			if strings.TrimSuffix(aws.StringValue(record.Name), ".") == resourceRecordSetName && aws.StringValue(record.Type) == "A" {
+				log.Info("Record already exists, skipping UPSERT.", "Domain", clusterDomain, "Record", aws.StringValue(record.Name))
+				recordExists = true
+				return false
+			}
+		}
+		return true
+	})
+
+	if err != nil || recordExists {
+		return err
+	}
```

Comment:
It would be nice to put this into a method that can be called to "ask" if a given record exists (with a given value?) so that it can be tested in isolation. Speaking of tests, it'd be nice to have tests to exercise the behaviour of this piece of code.

Some of the cases I'm curious about: What is the expected behaviour if the RRTYPE isn't `A`? How does your code change the current behaviour of `upsertARecord` changing the value of the record if it differs from what we want?

Reply:
Thanks for the review, I updated the PR to reflect your comments. 
As for the behaviour when RRTYPE doesn't match (or any other field), the function will return that it didn't find a match and `upsertARecord` will run the ChangeResourceRecordSets UPSERT  as usual.